text/0680-implicit-injection-deprecation.md

```diff
@@ -0,0 +1,127 @@
+If a project has `ember-data` installed, an initializer at runtime registers the `@ember-data/store` on both
+Ember.Route and Ember.Controller factory objects.  This is a nice convenience for a project with many routes
+that are used for data fetching. However, it leads to many inconveniences that hinder learning and advancement
+of the framework. This includes:
+
+1. By entangling specific objects with the whole dependency tree, we might be preventing incremental adoption.
+   For example, `ember-data` has laid out its plan for adopting only specific packages in [Project Trim](https://github.com/emberjs/data/issues/6166).
+   However, because `store` is injected by default on all Ember.Route and Ember.Controller factory objects,
+   a user cannot easily opt out of `@ember-data/store`. This is a common problem space in other communities as well.
+   With previous JetBrains IDE installations, adding plugins required a reload. However, by removing a contructor function
+   that injected all the dependencies at once, they were able to avoid reload after installation of a plugin.
+
+2. Eager initialization of the dependency tree from which the property came from prevents tree shaking and incurs a
+   performance hit. For example, if you app does not need `ember-data` to render the entry route, the `store` injection will
+   take up more CPU cycles than necessary, hurting common user facing metrics.
+
+3. Eager initialization makes it hard to users to write tests that need to take advantage of a stub.
+   This is often a silent and annoying hurdle that users have to get around when writing tests.
+   Moreover, this is especially apparent for users moving to Ember's new testing APIs in [RFC 232](https://github.com/emberjs/rfcs/blob/master/text/0232-simplify-qunit-testing-api.md) and [RFC 268](https://github.com/emberjs/rfcs/blob/master/text/0268-acceptance-testing-refactor.md),
+   ensuring instance initializers run before each test.
+
+4. Injection on commonly used framework objects can lead to deviation of project specific styleguides. For instance,
+   injecting the `router` on the Ember.Component factory object will certainly lead to instances where the `router` is explicitly
+   injected and other instances where is it implicitly injected.
+
+5. Users have come to associate `Route.model()` with a hook that returns `ember-data` models in the absense of explicit injection.
+   While this can be true, it is not wholly true. New patterns of data loading are becoming
+   accepted in the community including opting to fetch data in a Component.
+
+
+## Detailed design
+
+Removing implicit injections is not possible until we first issue deprecations.
+This helps us maintain our strict semver commitment to not completing a major version bump and
+removing APIs without a deprecation path. As a result, it will take us 3 phases to remove `owner.inject`.
+
+### 1. Deprecate implicit injection on target object
```

Comment:
Should mention _why_ we deprecate this first, and how you avoid the deprecation. Specifically, this would allow users to add `@service store;` explicitly as a way to prevent the deprecation and avoid relying on the implicit injection.